### PR TITLE
Dropped PHP 8.0 support, updated composer packages for PHPUnit 10

### PR DESCRIPTION
**Pull request by SuperDJ**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

- Changed PHP requirements;
- Added PHPUnit 10 (should fix https://github.com/laminas/laminas-xmlrpc/pull/33, https://github.com/laminas/laminas-xmlrpc/issues/24);

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -5,3 +5,4 @@
 /phpunit.xml
 /vendor/
 .phpcs-cache
+composer.lock
```
Comment (Ocramius) [inline .gitignore]: Should **not** be ignored

File: composer.lock [hunk 2]
```diff
@@ -1,4862 +0,0 @@
-{
```
Comment (Ocramius) [inline composer.lock]: This file must be preserved

File: composer.json [hunk 3]
```diff
@@ -17,17 +17,14 @@
...
-        },
         "allow-plugins": {
             "dealerdirect/phpcodesniffer-composer-installer": true
         }
     },
     "extra": {
     },
     "require": {
-        "php": "~8.0.0 || ~8.1.0 || ~8.2.0",
+        "php": "^8.0",
```
Comment (Ocramius) [inline composer.json]: We don't know if PHP 8.3 will break BC, so we cannot be this lax with the range.

The range should be `~8.1.0 || ~8.2.0`

File: composer.json [hunk 4]
```diff
@@ -37,7 +34,7 @@
     },
     "require-dev": {
         "laminas/laminas-coding-standard": "~2.5.0",
-        "phpunit/phpunit": "^9.5.27",
+        "phpunit/phpunit": "^9.5.27 || ^10.0",
```
Comment (Ocramius) [inline composer.json]: We should bump to `^10.0.0` directly here

File: composer.json [hunk 5]
```diff
@@ -17,17 +17,14 @@
     },
     "config": {
         "sort-packages": true,
-        "platform": {
-            "php": "8.0.99"
-        },
```
Comment (Ocramius) [inline composer.json]: We should put `8.1.99` here, and drop PHP 8.0 support

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: src/Generator/GeneratorInterface.php [hunk 6]
```diff
@@ -40,5 +40,5 @@ public function stripDeclaration($xml);
 
     public function flush();
 
-    public function __toString();
+    public function __toString(): string;
```
Comment (Ocramius) [inline src/Generator/GeneratorInterface.php:43]: Had to check these again: apparently adding `:string` (return type) is not a BC break on `__toString()`, thanks to the implicit `Stringable` implementation.

In fact, following is legal PHP 8.1 code:

```php
<?php

class Foo {
    function __toString(): string {}
}

class Bar extends Foo {
    function __toString() {}
}
```

**Review by Ocramius — APPROVED** [review]
Thanks @SuperDJ!

File: src/Generator/GeneratorInterface.php (hunk 6 above)
Comment (SuperDJ) [reply src/Generator/GeneratorInterface.php:43]: It was also required for Psalm to pass

**Pull request merged**